### ISSUE TITLE
Can't get the program to work.

### ISSUE THREAD

**Issue by johanhelmuth** [aikikode/fileshare#20]
OS: Elementary OS 5.0 Juno (based on Ubuntu 18.04)

I have Python2 and Python3 installed.

Not sure if this is a keybind issue or not, but I installed _"indicator-fileshare-0.6.0.deb"_ using `apt-get install`, it installed without any issues.

Problem is that nothing happens when I try to start the Fileshare Applet, no indicator that it has started or anything. I also checked if python3 could import Keybinder properly from gi.repository - it couldn't.

```
>>> from gi.repository import Keybinder
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 888, in _find_spec
AttributeError: 'DynamicImporter' object has no attribute 'find_spec'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/gi/importer.py", line 127, in find_module
    'introspection typelib not found' % namespace)
ImportError: cannot import name Keybinder, introspection typelib not found
```
But yeah, nothing happens when I click the applet, and <Super>P isn't responding, obviously.

**Comment by aikikode** [comment]
Hi @johanhelmuth,

thank you for the feedback. Sorry that you experience this issue.
Could you do the following, please:
1. Exit/quit the `indicator-fileshare` applet
2. Start it from the terminal as a normal user (not root)
```
$ indicator-fileshare
```
3. Try to click the applet icon (if it starts) and take a screenshot
4. Post here all the logs/traces from the terminal

That would help me a lot to find the cause of the issue. Thanks again for the report!

**Comment by johanhelmuth** [comment]
Hey @aikikode,

I just went through the steps of which you asked. I made sure that there was no instance of indicator-fileshare running. I then in the terminal typed in (non-root):

> $ indicator-fileshare

What happens is that the terminal gets stuck in loading the applet, I think. Basically I get an empty line below what I typed in, in the terminal. Ctrl+C couldn't even end the process in the terminal, so I had to force close the terminal.

Just to make it clear, because I'm terrible at explaining things, hehe - nothing pops up after I try to start the applet via the terminal, the terminal just get "stuck", not producing any output or anything.

Also, I'm happy to help! I'm stoked that you have created something similar to ShareX on Linux, can't wait to use it! :)

**Comment by aikikode** [comment]
Well, no output is also a symptom :) Could you check then if there's a file
```
~/.cache/indicator-fileshare.log
```
That's where the app writes its log. If it exists, could you paste its contents here, please?

**Comment by johanhelmuth** [comment]
> [MainThread] 2019-03-19 13:37:01,095 - INFO - ------------------------------
> [MainThread] 2019-03-19 13:37:01,095 - INFO - Started Fileshare Indicator
> [MainThread] 2019-03-19 13:37:01,095 - INFO - Fileshare Indicator version 0.6.0
> [MainThread] 2019-03-19 13:37:01,096 - DEBUG - SingleInstance: writing new pid 2473
> [MainThread] 2019-03-19 13:37:01,096 - ERROR - SingleInstance: exception while renaming '/tmp/tmplp49cpzb' to '/tmp/indicator-fileshare-johan.pid':
>  a bytes-like object is required, not 'str'
> [MainThread] 2019-03-19 13:37:01,096 - DEBUG - Indicator: creating
> [MainThread] 2019-03-19 13:37:01,097 - ERROR - Error: No section: 'SERVICE'

Here you go! I get the same output as root and non-root.

**Comment by aikikode** [comment]
Hi @johanhelmuth, so I installed Juno and investigated the issue. I was able to reproduce the same behaviour you described.

Elementary OS uses GTK-based Pantheon desktop environment. But it uses custom widget bar at the top of the screen (where fileshare should normally appear in other OS) called `wingpanel`.

The issue is that it doesn't allow pygtk apps (such as `fileshare`) to appear there. Or at least I couldn't find a way to do that without breaking the app for other OS. As far as I see some older versions of Elementary OS did allow that but not the latest one. I found a workaround for that including rebuilding "fixed" version of `wingpanel` to turn this functionality on again, but I don't think it's worth it.

Long story short there's no easy fix for this in Elementary OS now. I'm going to mark this issue as `bug` until either Elementary OS is fixed or I find a way to overcome this e.g. by making `fileshare` a standalone app instead of `appindicator` for Juno.

**Comment by johanhelmuth** [comment]
@aikikode I see! Thanks for investigating it, really appreciate it. I will wait patiently then!
